src/examples/checkpoints.ts

```diff
@@ -0,0 +1,83 @@
+import { CalendarUnit } from '@polymathnetwork/polymesh-sdk/types';
+
+import { getClient } from '~/common/client';
+
+/* 
+  This script showcases Checkpoints related functonality. It:    
+    - Creates a Checkpoint
+    - Fetches asset's Checkpoints
+    - Fetches Checkpoint details
+    - Creates a Schedule
+    - Fetches asset's Schedules
+    - Fetches Schedule details
+    - Fetches Checkpoints originated by a Schedule
+    - Deletes a Schedule
+*/
+(async (): Promise<void> => {
+  console.log('Connecting to the node...\n\n');
+  const api = await getClient(process.env.ACCOUNT_SEED);
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const identity = (await api.getCurrentIdentity())!;
+  console.log(`Connected! Current identity ID: ${identity.did}`);
+
+  const ticker = process.argv[2];
+
+  if (!ticker) {
+    throw new Error('Please supply a ticker as an argument to the script');
+  }
+
+  const token = await api.getSecurityToken({ ticker });
+  console.log(`Security Token found! Current token name is: ${(await token.details()).name}`);
+
+  const createQ = await token.checkpoints.create();
+  const newCheckpoint = await createQ.run();
+
+  console.log('New checkpoint has been created:');
+  console.log(`- Id: ${newCheckpoint.id}`);
+  console.log(`- Ticker: ${newCheckpoint.ticker}`);
+  console.log(`- Created at: ${await newCheckpoint.createdAt()}`);
+  console.log(`- Total supply: ${await newCheckpoint.totalSupply()}`);
```

Comment:
These two can be fetched in parallel with Promise.all

(Also I personally try to avoid inline awaits, it hurts readability)